dace/transformation/subgraph/subgraph_fusion.py

```diff
@@ -506,19 +639,17 @@ def redirect(redirect_node, original_node):
 
         # do a full global search and count each data from each intermediate node
         scope_dict = graph.scope_dict()
-        for state in sdfg.nodes():
-            for node in state.nodes():
-                if isinstance(
-                        node,
-                        nodes.AccessNode) and node.data in data_intermediate:
-                    # add them to the counter set in all cases
-                    data_counter[node.data] += 1
-                    # see whether we are inside the subgraph scope
-                    # if so, add to data_counter_subgraph
-                    # DO NOT add if it is in out_nodes
-                    if state == graph and \
-                       (node in intermediate_nodes or scope_dict[node] in map_entries):
-                        data_counter_subgraph[node.data] += 1
+        for node, state in sdfg.all_nodes_recursive():
```

Comment:
that does not make a lot of sense, in nested sdfgs you can have nodes with the same name but pointing to different data. please only loop over nodes in the current sdfg (all states though).